### ISSUE TITLE
Name missing for beta catenin (and other human proteins)?

### ISSUE THREAD

**Issue by cmungall** [geneontology/amigo#240]
http://amigo.geneontology.org/amigo/gene_product/UniProtKB:P35222

```
Name(s)    (n/a)
```

This is not a problem with the data
```
$ gzip -dc gene_association.goa_human.gz  | grep '^UniProtKB\tP35222' | cut -f1-3,10 | head
UniProtKB       P35222  CTNNB1  Catenin beta-1
UniProtKB       P35222  CTNNB1  Catenin beta-1
UniProtKB       P35222  CTNNB1  Catenin beta-1
UniProtKB       P35222  CTNNB1  Catenin beta-1
UniProtKB       P35222  CTNNB1  Catenin beta-1
UniProtKB       P35222  CTNNB1  Catenin beta-1
UniProtKB       P35222  CTNNB1  Catenin beta-1
UniProtKB       P35222  CTNNB1  Catenin beta-1
UniProtKB       P35222  CTNNB1  Catenin beta-1
UniProtKB       P35222  CTNNB1  Catenin beta-1
```

`Catenin beta-1` should be in the name field. As well as a display issue, this is a search issue, it's impossible to find this gene product unless you have memorized the symbol.

This is *partly* due to the fact uniprot doesn't include useful symbols:

```
UniProtKB       P35222  CTNNB1  Catenin beta-1  CTNB1_HUMAN|CTNNB1|CTNNB|OK/SW-cl.35|PRO2286
UniProtKB       P35222  CTNNB1  Catenin beta-1  CTNB1_HUMAN|CTNNB1|CTNNB|OK/SW-cl.35|PRO2286
UniProtKB       P35222  CTNNB1  Catenin beta-1  CTNB1_HUMAN|CTNNB1|CTNNB|OK/SW-cl.35|PRO2286
UniProtKB       P35222  CTNNB1  Catenin beta-1  CTNB1_HUMAN|CTNNB1|CTNNB|OK/SW-cl.35|PRO2286
UniProtKB       P35222  CTNNB1  Catenin beta-1  CTNB1_HUMAN|CTNNB1|CTNNB|OK/SW-cl.35|PRO2286
UniProtKB       P35222  CTNNB1  Catenin beta-1  CTNB1_HUMAN|CTNNB1|CTNNB|OK/SW-cl.35|PRO2286
UniProtKB       P35222  CTNNB1  Catenin beta-1  CTNB1_HUMAN|CTNNB1|CTNNB|OK/SW-cl.35|PRO2286
UniProtKB       P35222  CTNNB1  Catenin beta-1  CTNB1_HUMAN|CTNNB1|CTNNB|OK/SW-cl.35|PRO2286
UniProtKB       P35222  CTNNB1  Catenin beta-1  CTNB1_HUMAN|CTNNB1|CTNNB|OK/SW-cl.35|PRO2286
UniProtKB       P35222  CTNNB1  Catenin beta-1  CTNB1_HUMAN|CTNNB1|CTNNB|OK/SW-cl.35|PRO2286
```

Most likely golr loader issue but we need to keep a ticket here

**Comment by cmungall** [comment]
False positive. Symbol/name confusion

**Comment by kltm** [comment]
Hmmm...possibly clobbering somewhere in the load?

```
sjcarbon@moiraine:~/local/src/svn/geneontology.org/trunk/gene-associations$:)  zgrep -H -P "^UniProtKB\tP35222" *.gz | cut -f1-3,10 | uniq
gene_association.goa_human.gz:UniProtKB	P35222	CTNNB1	Catenin beta-1
gene_association.goa_ref_human.gz:UniProtKB	P35222	CTNNB1	Catenin beta-1
gene_association.goa_uniprot_noiea.gz:UniProtKB	P35222	CTNNB1	Catenin beta-1
gene_association.reactome.gz:UniProtKB	P35222	CTNB1_HUMAN	
```

Different symbol there...

**Comment by cmungall** [comment]
Ahh reactome clobbering, sure we have a ticket somewhere.

We should pass this upstream but at the same time be defensive

**Comment by kltm** [comment]
Just to confirm this is at least loader or upstream, from AmiGO:
```json
      {
        "document_category":"bioentity",
        "id":"UniProtKB:P35222",
        "bioentity":"UniProtKB:P35222",
        "bioentity_internal_id":"P35222",
        "bioentity_label":"CTNNB1",
        "bioentity_name":"",
        "source":"UniProtKB",
        "type":"protein",
        "taxon":"NCBITaxon:9606",
        "taxon_label":"Homo sapiens",
        ...
}
```

Okay, an easy to implement hotfix then.

**Comment by kltm** [comment]
See #211.

**Comment by kltm** [comment]
@stuartmiyasato confirms that reactome is not being loaded on production; need to look for problem in loader then.

**Comment by cmungall** [comment]
Seth, if you used the most up to date svn here https://github.com/geneontology/amigo/issues/240#issuecomment-136743464

Then reactome is the *only* explanation for this. Can we see the production logs?

**Comment by kltm** [comment]
It also /feels/ like it should be reactome to me, but hard evidence seems to indicate otherwise (so far). I'm rerunning in two different places: a full 2.3.x default load on nakama and just the two loaded GAFs that contain UniProtKB:P35222 (and not the two we don't load) locally. Unfortunately, I'll have nothing for a couple more hours (and tie up my laptop, hello from the github app).

**Comment by kltm** [comment]
And yes, the full load is off the purls and the local load is on a refreshed svn.

**Comment by stuartmiyasato** [comment]
I did a "grep -i reactome" on the load log file and got the following:

golr> grep -i reactome golr-load-20150831
Generating /data/src/svn/owltools/OWLTools-Oort/target/apidocs/owltools/ontologyrelease/reports/go/External2GoReports.External2GoReactome.html...
Generating /data/src/svn/owltools/OWLTools-Oort/target/apidocs/owltools/ontologyrelease/reports/go/class-use/External2GoReports.External2GoReactome.html...

So I don't think that's helpful...

The entire log file is 1.9MB.  I don't see a way of attaching that as a text file.  @cmungall and @kltm , would you like this file in email?  What is the best way to get that to you?

**Comment by cmungall** [comment]
I have the log, the following are loaded:
```
http://www.geneontology.org/gene-associations/gene_association.GeneDB_Lmajor.gz
http://www.geneontology.org/gene-associations/gene_association.GeneDB_Pfalciparum.gz
http://www.geneontology.org/gene-associations/gene_association.GeneDB_Spombe.gz
http://www.geneontology.org/gene-associations/gene_association.GeneDB_Tbrucei.gz
http://www.geneontology.org/gene-associations/gene_association.PAMGO_Atumefaciens.gz
http://www.geneontology.org/gene-associations/gene_association.PAMGO_Ddadantii.gz
http://www.geneontology.org/gene-associations/gene_association.PAMGO_Mgrisea.gz
http://www.geneontology.org/gene-associations/gene_association.PAMGO_Oomycetes.gz
http://www.geneontology.org/gene-associations/gene_association.aspgd.gz
http://www.geneontology.org/gene-associations/gene_association.cgd.gz
http://www.geneontology.org/gene-associations/gene_association.dictyBase.gz
http://www.geneontology.org/gene-associations/gene_association.ecocyc.gz
http://www.geneontology.org/gene-associations/gene_association.fb.gz
http://www.geneontology.org/gene-associations/gene_association.goa_chicken.gz
http://www.geneontology.org/gene-associations/gene_association.goa_cow.gz
http://www.geneontology.org/gene-associations/gene_association.goa_dog.gz
http://www.geneontology.org/gene-associations/gene_association.goa_human.gz
http://www.geneontology.org/gene-associations/gene_association.goa_pig.gz
http://www.geneontology.org/gene-associations/gene_association.goa_uniprot_noiea.gz
http://www.geneontology.org/gene-associations/gene_association.gonuts.gz
http://www.geneontology.org/gene-associations/gene_association.gramene_oryza.gz
http://www.geneontology.org/gene-associations/gene_association.jcvi.gz
http://www.geneontology.org/gene-associations/gene_association.mgi.gz
http://www.geneontology.org/gene-associations/gene_association.pombase.gz
http://www.geneontology.org/gene-associations/gene_association.pseudocap.gz
http://www.geneontology.org/gene-associations/gene_association.rgd.gz
http://www.geneontology.org/gene-associations/gene_association.sgd.gz
http://www.geneontology.org/gene-associations/gene_association.sgn.gz
http://www.geneontology.org/gene-associations/gene_association.tair.gz
http://www.geneontology.org/gene-associations/gene_association.wb.gz
http://www.geneontology.org/gene-associations/gene_association.zfin.gz
http://www.geneontology.org/gene-associations/submission/paint/pre-submission/gene_association.paint_other.gaf
```

All appear to be legit non-clobberers.. the mystery deepens.

**Comment by cmungall** [comment]
Also, if it were reactome clobbering, then the symbol would have `_HUMAN` suffixed. So the primary hypothesis now switches to golr loader/owltools problem

**Comment by cmungall** [comment]
OK, getting weirder. 

On labs and prod-test:
http://amigo2.berkeleybop.org/amigo/gene_product/UniProtKB:P35222
http://amigo2-test.stanford.edu/amigo/gene_product/UniProtKB:P35222
The name field in the summary (presumably bioentity doc) says n/a, but in the 2nd col of the table (presumably association doc), it is populated

On production:
http://amigo.geneontology.org/amigo/gene_product/UniProtKB:P35222
In *neither*

**Comment by ValWood** [comment]
fuck, what? thats 5 years old....
http://www.geneontology.org/gene-associations/gene_association.GeneDB_Spombe.gz

**Comment by ValWood** [comment]
Ah I thought that was panther... but that file should no longer exist *anywhere*. I am sure we got everything updated to remove GeneDB_Spombe....

**Comment by cmungall** [comment]
The file doesn't exist anymore and pombase is in the list, so apparently it's harmless but it is confusing, and it's worrying that it doesn't cause the loader to barf and does speak to the fact that we need more metadata love. We need a central yaml file or similar on what the core sets of GAFs are for GO

**Comment by kkarra** [comment]
It looks like the results order is different between labs/prod-test vs production. In GONUTS source name does not appear.

**Comment by stuartmiyasato** [comment]
Turns out that there's a data discrepancy between Stanford's test and production environments.  The test data is slightly older than production.  Maybe this will help us track down the issue.  We will update the data on test later today with the current production data (from the 8/31 load) and see if the 2nd column remains populated or not.

**Comment by kkarra** [comment]
I will go ahead and update the index files on amigo2-test with 8/31 load files now.

**Comment by kltm** [comment]
Okay great--I'm rather at my wits end about not being able to load myself; fingers crossed for a transient problem.

**Comment by stuartmiyasato** [comment]
As a side note...  I have disabled the daily GOlr load.  However, I am running a load manually (but not pushing the indexes anywhere).  Last night's load failed as has a number of previous loads in the past month or so.  I was wondering if those failures might be related.  The current load looks like it will complete successfully -- I kicked it off around noon and it's in the optimization stage, so I'm not seeing the 14 hour load that @kltm mentioned elsewhere.

Hmm, wondering if I should have commented in #241 instead of here...

**Comment by stuartmiyasato** [comment]
Still off-topic, but since I commented here earlier...

I spoke too soon...  The load failed, but I think it's because I forgot to stop jetty before starting the load.

Okay, future comments will be more on-topic for this thread.

**Comment by stuartmiyasato** [comment]
I have loaded the 8/31 data on amigo2-test and am now seeing the same results as in production.

But on that note, no one has commented on Kalpana's comment above re: GONUTS.  On all three servers where "Assigned by" is GONUTS, we lose the "Catenin beta-1" under the Gene/product name column, even on the berkeleybop.org server.

I searched for CTNNB1 in the GONUTS GAF file and got these lines:

UniProtKB	P35222	CTNNB1		GO:0019900	PMID:20307497	IPI	UniProtKB:Q9H2X6	F			protein	taxon:9606	20101107	GONUTS		
UniProtKB	P35222	CTNNB1		GO:0005634	PMID:20307497	IDA	protein	taxon:9606	20101107	GONUTS		
UniProtKB	P35222	CTNNB1		GO:0005737	PMID:20307497	IDA	protein	taxon:9606	20101107	GONUTS		
UniProtKB	P35222	CTNNB1		GO:0016310	PMID:20307497	IMP	protein	taxon:9606	20101107	GONUTS

Does this look correct?  (I have no idea...)

**Comment by kltm** [comment]
Hm. If that's the whole line, that's not good--I believe the "Catenin beta-1" label should be in there. If we were loading these, they could be the ones clobbering the other entries, but I don't think we're actually loading anything from GONUTS.

**Comment by cmungall** [comment]
Are the go nuts assocs in another gaf?

**Comment by kltm** [comment]
Shouldn't we find them then when we grep?

**Comment by kltm** [comment]
Okay, finally have gotten to the point where I can look at this stuff again. So, loading a single GAF locally, the problem does not occur. This means that it is either: 1) a clobber happening somewhere along the way with conflicting GAFs (something that seems unlikely after grepping) or 2) a hiccup in owltools that was fixed after the last update. I'll do an update and a full load on a dev server and see if the problem persists.

**Comment by stuartmiyasato** [comment]
So elaborating a bit on my previous comment -- I again did a grep on CTNNB1 in the gene_association.gonuts file and got the four lines above.  I did a case-insensitive grep on "catenin beta-1" and got nothing.

http://amigo.geneontology.org/amigo/load_details does list http://www.geneontology.org/gene-associations/gene_association.gonuts.gz as a file being loaded.  Did we already eliminate the GONUTS file a possibly being a culprit?

**Comment by kltm** [comment]
Sorry, I didn't read your previous comment correctly. If you are loading gonuts, that would be an issue. Possibly /the/ issue.

Going with that, the question is either: should we drop GONUTS, fix it, or should we make the loader more defensive (which we should, but should that be the fix for the immediate problem)?

**Comment by stuartmiyasato** [comment]
I could run a load where we ignore the GONUTS file and load the indexes on amigo2-test.  However, with regards to #241 , would I need to upgrade OWLTools or anything else on the loading server first?

**Comment by kltm** [comment]
Yes, for #241, owltools needs to be pulled to the latest and rebuilt. However, in doing that, there needs to be schema updates because we don't have a separate owltools release (related to an old upstream fix for #209 that was not supposed to be in the release).

Because of these interdependencies, I've "restarted" the 2.3 release process and will close the remaining issues (and test the fixes here) and merge from master before pinging you again. You'd need to: 1) update/rebuild owltools, 2) pull 2.3.x fresh and build, 3) update the solr schema with the one supplied by 2.3.x and restart solr, and 4) start load again.

**Comment by kltm** [comment]
@cmungall The most correct immediately solution would seem to me to be get the gonuts GAFs corrected upstream, leaving the fixes in owltools as a separate issue (owlcollab/owltools#134).

Barring that, I'd suggest dropping gonuts temporarily until the data is fixed. If you can't get back to me, I'll reach out to Jim Hu and see if this can be cleared-up, and then have @stuartmiyasato drop it from the load so we can move forward on 2.3.

**Comment by jimhu-tamu** [comment]
Sorry about the GONUTS problems. I think that gene_association.gonuts.gz could just be deleted - We haven't been maintaining that file and I hadn't even realized it was still in the submission directory. I think it's a remnant of some early attempt to submit CACAO annotations where I'd be hard pressed to reconstruct the code used to construct the annotations. It hasn't been updated since Oct 2012 based on the timestamp in my copy of the svn gene_associations/submission directory and skimming the dates of the annotations it's probably older than that. 

We've been sending the CACAO annotations via Protein2GO with the source indicated as CACAO.

**Comment by kltm** [comment]
Okay, thank you for the information. In that case, I'll go ahead and remove the file from the loader configs and we'll restart the release on Monday. I'll close when amigo2-test at Stanford passes all tests, including the new ones for #248.

**Comment by kltm** [comment]
I can now confirm that the issue seems to be fixed on dev (finally!).
